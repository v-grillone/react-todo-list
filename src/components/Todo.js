import React from "react";

function Todo({ todo, completeTask }) {
    function completeClick(e) {
        completeTask(todo.id);
        const btnText = e.target.innerText
        btnText === 'Complete' ? e.target.innerText = 'Uncomplete' : e.target.innerText = 'Complete';
    }
    const { id, title, completed } = todo;
    const h1 = <h1 className="text-2xl w-52">{title.toUpperCase()}</h1>;
    const text = completed ? <strike>{h1}</strike> : h1;
    const btnCompleted = <button className='px-2 border-2 rounded border-slate-500 text-white bg-slate-400 hover:bg-slate-400' onClick={completeClick}>Complete</button> 
    return <div data-testid={`todo-${id}`} className='task flex flex-row'>{text}{btnCompleted}</div>
}

export default Todo;