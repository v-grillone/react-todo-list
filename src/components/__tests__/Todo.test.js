import { render, screen , cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Todo from '../Todo';

afterEach(() => {
    cleanup();
})

test('Should render non-completed todos', () => {
    const todo = {id: 1, title: 'Gym', completed: false};
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('GYM');
    expect(todoElement).not.toContainHTML('<strike>');
});

test('Should render completed todos', () => {
    const todo = {id: 2, title: 'Homework', completed: true};
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('HOMEWORK');
    expect(todoElement).toContainHTML('strike');
});

test('matches snapshot', () => {
    const todo = {id: 1, title: 'Gym', completed: false};
    const tree = renderer.create(<Todo todo={todo}/>).toJSON();
    expect(tree).toMatchSnapshot();
})