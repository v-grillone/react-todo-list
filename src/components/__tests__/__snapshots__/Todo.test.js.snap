// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="task flex flex-row"
  data-testid="todo-1"
>
  <h1
    className="text-2xl w-52"
  >
    GYM
  </h1>
  <button
    className="px-2 border-2 rounded border-slate-500 text-white bg-slate-400 hover:bg-slate-400"
    onClick={[Function]}
  >
    Complete
  </button>
</div>
`;
